### DISCUSSION TITLE
How to make pnpm show warning during installation?

### DISCUSSION THREAD

**Discussion by sindzicat** [pnpm/pnpm#4217] [Q&A] [answered]
Hello!

I use node.js 16 on Windows 10.

When I install gulp using pnpm I see the following:

```
λ pnpm add gulp
Packages: +294
++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
Packages are hard linked from the content-addressable store to the virtual store.
  Content-addressable store is at: P:\.pnpm-store\v3
  Virtual store is at:             node_modules/.pnpm
Progress: resolved 298, reused 295, downloaded 2, added 294, done

dependencies:
+ gulp 4.0.2
Already up-to-date
Progress: resolved 1, reused 1, downloaded 0, added 0, done
```

But when I use npm to install gulp I see additional info:

```
λ npm install gulp
npm WARN deprecated chokidar@2.1.8: Chokidar 2 will break on node v14+. Upgrade to chokidar 3 with 15x less dependencies.
npm WARN deprecated resolve-url@0.2.1: https://github.com/lydell/resolve-url#deprecated
npm WARN deprecated urix@0.1.0: Please see https://github.com/lydell/urix#deprecated
...
```

I see, that `gulp.watch()` won't work on Node.js 16 because gulp still uses chokidar 2.

I used pnpm to install gulp and didn't see such message. As result I was need a whole week to find why `gulp.watch()` don't work.

How to turn on such warnings in pnpm?

**Comment by zkochan** [comment]
Currently pnpm prints only deprecation warnings of direct dependencies. But all the deprecation messages are found in the lockfile. So you could search for all "deprecated" in `pnpm-lock.yaml`.

I guess if we don't print deprecation messages of subdeps, we should at least print that there are such packages and the information about them is in the lockfile.

Maybe we could print something like this in the installation summary: `there are x deprecated packages in your dependencies, search for "deprecated" in pnpm-lock.yaml to see them`

**Reply by sindzicat** [reply]
Sorry, but this means that one more steps needs to see what's deprecated.

**Reply by zkochan** [reply]
The latest pnpm prints all the deprecation warnings.

**Reply by sindzicat** [reply]
> The latest pnpm prints all the deprecation warnings.

Just tested. Yes, you're right. And this is very cool now!
Many thanks!

**Reply by kjkent** [reply]
It's great that pnpm now shows deprecated subdependencies when running `pnpm up`; however, it doesn't say which direct dependency is pulling in the outdated subdependencies. As well as this, `pnpm outdated` returns nothing for this, as does `pnpm i`.

As far as I can tell, the only way to find which direct dependencies are responsible is to run `pnpm up`, then `pnpm ls --depth <unknown number>`, grep the tree for each of the outdated deps, then follow the tree back up to the direct dependency that pulled them in.

Unless there's an easier way?  There's an issue (https://github.com/pnpm/pnpm/issues/7323) regarding this, but, I've also seen https://github.com/pnpm/pnpm/issues/7339, which suggests an issue with `pnpm outdated` regarding pre-release tags (the package that's got the outdated dep in my case is `@11ty/eleventy@3.0.0-alpha-14`), though I'm not sure if this is related

**Comment by zkochan** [comment, marked as answer]
Fix released in [6.26.1](https://github.com/pnpm/pnpm/releases/tag/v6.26.1)

**Reply by harisraharjo** [reply]
I can confirm that not all warning are shown. I have total 8 warnings, 3 warnings are hidden with a "and 3 other warnings" message. How to show all of those warnings?
